src/main/java/generic/fixture/GenericFixture.java

```diff
@@ -532,35 +557,46 @@ private static Object getRandomForType(Class<?> fieldType,
 
             assert map != null;
 
-            for (int i = 0; i < 1; i++) {
-                Object key = getObjectByClass(innerClasses[0], customFields, currentPath);
-                Object value = getObjectByClass(innerClasses[1], customFields, currentPath);
+            for (int i = 0; i < numberOfItens; i++) {
+                Object key = getObjectByClass(innerClasses[0], customFields, currentPath, numberOfItens);
+                Object value = getObjectByClass(innerClasses[1], customFields, currentPath, numberOfItens);
                 tryPutOnMap(type, map, key, value);
             }
 
             return map;
         }
 
         if (fieldType.isArray()) {
-            Object array = Array.newInstance(fieldType.getComponentType(), 1);
-            if(implementsMap(fieldType.getComponentType()) || isDictionary(fieldType.getComponentType())) {
-                Type typeMap = (((GenericArrayType) type).getGenericComponentType()); //This cast is necessary to parse Map<key,value>[] to Map<key,value>
-                Array.set(array, 0, getRandomForType(fieldType.getComponentType(), typeMap, hashMap, customFields, currentPath));
-            } else {
-                Array.set(array, 0, getObjectByClass(fieldType.getComponentType(), customFields, currentPath));
+
+            Class<?> arrayType = fieldType.getComponentType();
+            Object array = Array.newInstance(arrayType, numberOfItens);
+
+            for (int i = 0; i < numberOfItens; i++) {
+                if (hasTypeParameters(arrayType)) {
+                    //This cast is necessary to parse Map<K,V>[] to Map<K,V>, os List<E>[] to List<E>
```

Comment:
//This cast is necessary to parse Map<K,V>[] to Map<K,V>, or List<E>[] to List<E>